src/main/java/org/jenkinsci/test/acceptance/po/JobsMixIn.java

```diff
@@ -35,7 +35,11 @@ public JobsMixIn(ContainerPageObject context) {
             }
         }).click();
 
-        clickButton("OK");
+        //Sometimes we need to give some time to ok button to become enabled
+        waitTillEnabledAndClick("OK", 10, TimeUnit.SECONDS);
+
+        //sometimes job creation is not fast enough, so make sure it's finished before continue
+        waitFor(by.name("config"));
```

Comment:
@recena I doubt that changing that `waitFor` for some JS magic (that anyway would probably depend also on the UI or worst be browser dependant) would  make a test, that already depends upon a button with a certain caption to exist, more robust or proof safe. That said, I truly believe ids are one of the most robust selectors out there, if the test fails the error message will be very informative and the fail easy to track. 

Changing that selector would mean probably a lot of collateral changes to be done, like CSS rules, JS code and so on. On that context, changing the test should not be difficult. So I must agree with @amuniz here. 

The main problem here is not tests being more or less fragile, but the lack of communication between jenkinsci and the acceptance-test-harness when UI changes, unlike unit tests there is no compiler here, so there is no easy way to ensure changes in the UI are propagated to tests. Honestly I believe a PR on this project is not the place to try to solve that problem, and that a separate Jira ticket should be created to discuss the actual fragility of this suite in general 

@olivergondza The problem is that without that `waitFor` is the `create` method itself the one that randomly fails in low resources machines (8 GB RAM), resulting in jobs not being created and this particular test failing.